File: abril/19.04/restoring.cpp

#include<cstdlib>
#include<iostream>
#include<algorithm>
#include<vector>

/*
 * Desfio A - Restoring Three Numbers
 */
using namespace std;


int main(){
    vector <int> entrada(4);
    int a,b, c, i;

    for(i=0;i<4;i++)
        scanf("%d",&entrada[i]);
    
    sort(entrada.begin(), entrada.end());


    a = entrada[3] - entrada[2];
    b = entrada[3] - entrada[1];
    c = entrada[3] - entrada[0];

    printf("%d %d %d\n", a, b, c);


    return 0;
}

File: abril/19.04/restoring.py
'''
 Desafio A - Restoring Three Numbers
 '''

entrada = []
entrada = input()

entrada = entrada.split(" ")
entrada = list(map(int, entrada))  #inteiros
#print(entrada)

entrada.sort()

#print(entrada)

a = entrada[3] - entrada[2] 
b = entrada[0] - a
c = entrada[2] - b


print(c, b, a)




